dashboard/app/helpers/users_helper.rb

```diff
@@ -245,15 +240,67 @@ def user_summary(user)
         pages_completed.each_with_index do |result, index|
           progress["#{level_id}_#{index}"] = {
             result: result,
-            submitted: submitted ? true : nil,
-            readonly_answers: readonly_answers ? true : nil
+            submitted: level_progress[:submitted],
+            readonly_answers: level_progress[:readonly_answers]
           }.compact
         end
       end
     end
     progress
   end
 
+  # Summarizes a user's level progress for bubble choice level (parent level and sublevels)
+  private def get_bubble_choice_progress(level, user, user_levels_by_level, script_level, paired_user_levels, include_timestamp)
```

Comment:
NIT: in terms of figuring out how this code works, imo `get_level_progress` should come before this.

Reply:
that said, i do think factoring this out into its own helper function really improves the readability of `merge_user_progress_by_level` 👍

Reply:
I'll move it down!